### PR TITLE
(21.3) Mx builds should use python3 to avoid issues

### LINKED ISSUE #208: Make mandrel-packaging set python version to be used with mx to 3
As python 2 reached its end of life in Jan 2020, we should make sure we no longer use it when building mandrel. `mx` doesn't seem to enforce the use of python3 so if python2 is found in the path it might as well use it.

Related issues:
* https://github.com/graalvm/mx/issues/247
* https://github.com/graalvm/mx/issues/238

### PR DESCRIPTION
**Pull request by galderz**
Closes #208

### CODE REVIEW

File: build.java [hunk 1]
```diff
@@ -789,6 +795,7 @@ static void build(
             return Tasks.Exec.of(
                 args
                 , mandrelRepo.apply(Path.of("substratevm"))
+                , new EnvVar("MX_PYTHON", "python3")
```
Comment (zakkak) [inline build.java:798]: See https://github.com/graalvm/mandrel-packaging/pull/209#discussion_r761781529

**Comment by galderz** [comment]
@zakkak Updated PR.

**Review by zakkak — APPROVED** [review]

**Comment by Karm** [comment]
Note: This now requires `python3` to exist, even on a system where `python` actually means Python 3 and there is no Python 2.    So....I am adding

```
mklink C:\Python39\python3.exe C:\Python39\python.exe
```

to make it work now.   Not sure if build.java shouldn't be smarter about it.

**Comment by zakkak** [comment]
Good point.

> Not sure if build.java shouldn't be smarter about it.

It's probably worth checking if `python3` exists before setting it as the `MX_PYTHON`.

**Pull request merged**